include/openssl/evp.h

```diff
@@ -367,8 +369,9 @@ OPENSSL_EXPORT int EVP_DigestSign(EVP_MD_CTX *ctx, uint8_t *out_sig,
 // signing options.
 //
 // For single-shot signing algorithms which do not use a pre-hash, such as
-// Ed25519 and Dilithium, |type| should be NULL. The |EVP_MD_CTX| itself is
-// unused but is present so the API is uniform. See |EVP_DigestVerify|.
+// Ed25519, or when using ML-DSA in non pre-hash mode, |type| should be NULL.
```

Comment:
do we eventually plan to support the other digest algorithms allowed by FIPS 204? if so, when would be a good time to do that?

right now, we're basically committing to `type == NULL` resulting in a default of SHAKE-256. changing that behavior in the future (i.e. after this release) would be backwards-incompatible.

I'm not saying this is necessarily a bad thing, I just want to make sure we're consciously making that decision and committing to it.

On the other hand, if we require that SHAKE-256 be specified explicitly by the caller here, we're not committing to any such defaults, and can still defer other digest types to later releases.

Reply:
We can enforce SHAKE-256 when we implement HashML-DSA as this will be used by EVP_DigestSign and will not be precluded in `uses_prehash` as per the comment https://github.com/aws/aws-lc/pull/2113#discussion_r1916992591, while now Pure and External Mu ML-DSA (which is still pure) is precluded from `uses_prehash`.
I think that making that contract for pure Ed25519 and ML-DSA with MD being NULL is beneficial to say that these are special modes that don't adhere to the typical algorithms in which `uses_prehash` is true.